pkg/datapath/linux/node.go

```diff
@@ -1351,6 +1365,16 @@ func (n *linuxNodeHandler) NodeConfigurationChanged(newConfig datapath.LocalNode
 		}
 
 		if n.enableNeighDiscovery {
+			// store the neighDiscoveryLink used so that we can remove the ARP
+			// PERM entries created by this configuration run in case it's
+			// necessary
+			err := storeNeighLinks(option.Config.StateDir, []string{ifaceName})
```

Comment:
Let's move this after the netlink query below.

Reply:
Good point